Sources/CoreFoundation/CFPlatform.c

```diff
@@ -230,6 +234,29 @@ const char *_CFProcessPath(void) {
         char *res = realpath(ps->ps_argvstr[0], NULL);
         argv0 = res? res: strdup(ps->ps_argvstr[0]);
     }
+#elif defined(__FreeBSD__)
+    // see sysctl(3), pid == -1 means current process
+    int mib[4] = {CTL_KERN, KERN_PROC, KERN_PROC_PATHNAME, -1};
+    int sysctl_ret = 0;
+    size_t len = PATH_MAX + 1;
+    argv0 = calloc(len, 1);
+
+    sysctl_ret = sysctl(mib, 4, argv0, &len, NULL, 0);
+
+    // in case for whatever reason the path is > PATH_MAX
+    if (sysctl_ret == -1 && errno == ENOMEM) {
+        // get size needed
+        sysctl_ret = sysctl(mib, 4, NULL, &len, NULL, 0);
```

Comment:
Any reason to not do this unconditionally to compute the required length?

Reply:
Since we should almost never the case when exec  path is > PATH_MAX, so we can save an extra syscall in most cases. (In fact, I'm not sure if it's even possible on generic FreeBSD, but there are many FreeBSD variants floating around...)